### PR TITLE
interfaces: Add unsafe conversion primitives to LF spec

### PR DESCRIPTION
**Pull request by sofiafaro-da**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/spec/daml-lf-1.rst [hunk 1]
```diff
@@ -676,10 +676,12 @@ Then we can define our kinds, types, and expressions::
...
+       | 'from_interface' @τ₁ @τ₂ e                 -- ExpFromInterface: Turn an interface value back into a template value (returns optional) [Daml-LF ≥ 1.dev]
+       | 'unsafe_from_interface'  @τ₁ @τ₂ e₁ e₂     -- ExpUnsafeFromInterface: Turn an interface value back into a template value (throws fatal error) [Daml-LF ≥ 1.dev]
        | 'call_interface' @τ f e                    -- ExpCallInterface: Call a method on an interface value [Daml-LF ≥ 1.dev]
        | 'to_required_interface'  @τ₁ @τ₂ e         -- ExpToRequiredInterface: Upcast an interface value to an interface it requires [Daml-LF ≥ 1.dev]
-       | 'from_required_interface'  @τ₁ @τ₂ e       -- ExpFromRequiredInterface: Downcast an interface value to an interface that requires it [Daml-LF ≥ 1.dev]
+       | 'from_required_interface'  @τ₁ @τ₂ e       -- ExpFromRequiredInterface: Downcast an interface value to an interface that requires it (returns optional) [Daml-LF ≥ 1.dev]
+       | 'unsafe_from_required_interface'  @τ₁ @τ₂ e₁ e₂   -- ExpUnsafeFromRequiredInterface: Downcast an interface value to an interface that requires it (throws fatal error) [Daml-LF ≥ 1.dev]
```
Comment (akrmn) [inline daml-lf/spec/daml-lf-1.rst:684]: `r/mildlyinfuriating`

**Review by akrmn — APPROVED** [review]
thank you!

File: daml-lf/spec/daml-lf-1.rst (hunk 1 above)
Comment (akrmn) [reply daml-lf/spec/daml-lf-1.rst:684]: (I mean, the misaligned `--`, but there's nothing to do here, I'd rather keep the small diff)

**Review by remyhaemmerle-da — COMMENTED** [review]
Thanks.

**Pull request merged**